src/main/java/in/projecteka/gateway/clients/ServiceClient.java

```diff
@@ -67,11 +68,19 @@
     }
 
     private <T> Mono<Void> route(T request, String url, String routingKey, String clientId) {
-        return routingKey.equals(X_HIP_ID) || routingKey.equals(X_HIU_ID)
-                ? identityService.authenticate()
-                    .flatMap(token -> bridgeWebClientBuilder(request, url, token, routingKey, clientId)).then()
-                : identityService.authenticate()
-                    .flatMap(token -> cmWebClientBuilder(request, url, token)).then();
+
+        MDC.put("path", url);
+        MDC.put("method", "POST");
+        if (routingKey.equals(X_HIP_ID) || routingKey.equals(X_HIU_ID)) {
+            MDC.put("target-id", clientId);
+            logger.info("Target service info");
+            return identityService.authenticate()
+                    .flatMap(token -> bridgeWebClientBuilder(request, url, token, routingKey, clientId)).then();
+        }
```

Comment:
_**The MDC is managed on a per thread basis**_. A child thread automatically inherits a _copy_ of the mapped diagnostic context of its parent.

Reply:
I think this is what I mentioned using thread is not a right way to do it, and in case of Reactive programming, you have https://stackoverflow.com/questions/51185133/pass-a-context-from-one-flux-mono-to-another this, and it's kind of hard to get the head around it.

Reply:
I think this is what I mentioned using thread is not a right way to do it, and in case of Reactive programming, you have https://stackoverflow.com/questions/51185133/pass-a-context-from-one-flux-mono-to-another this, and it's kind of hard to get the head around it.